gnomad/utils/vep.py

```diff
@@ -720,12 +718,156 @@ def get_csq_from_struct(
+        transcripts before determining the most severe consequence. Default is False.
+    :param prioritize_loftee: Whether to prioritize LOFTEE consequences. Default is
+        False.
+    :param prioritize_loftee_no_flags: Whether to prioritize LOFTEE consequences with no
+        flags over those with flags. Default is False.
+    :param additional_order_field: Field name of the consequence annotation to use for
+        additional ordering to apply to the consequences in the list. Default is None.
+    :param additional_order: The ordering to use for prioritizing consequences in the
+        `additional_order_field`. Default is None.
+    :return: ArrayExpression with of the consequences that match the most severe
+        consequence.
+    """
+    if ((additional_order_field is None) + (additional_order is None)) == 1:
+        raise ValueError(
+            "If `additional_order_field` is provided, `additional_order` must also be"
+            " provided and vice versa."
+        )
+
+    loftee_labels = loftee_labels or LOFTEE_LABELS
+
+    if filter_to_protein_coding:
+        logger.info("Filtering to protein-coding transcripts...")
+        csq_expr = filter_vep_transcript_csqs_expr(csq_expr, protein_coding=True)
+
+    def _filter_to_most_severe(
+        expr: hl.expr.ArrayExpression,
+        field: str,
+        order: Optional[List[str]] = None,
+    ) -> Tuple[hl.expr.StringExpression, hl.expr.ArrayExpression]:
+        """
+        Filter to the most severe consequence in a list of consequences.
+
+        :param expr: ArrayExpression of consequences to filter.
+        :param field: Field name of the consequence annotation.
+        :param order: List indicating the order of VEP consequences, sorted from high to
+            low impact. Default is None, which uses the value of the `CSQ_ORDER` global.
+        :return: The most severe consequence and an ArrayExpression of consequences that
+            match the most severe consequence.
+        """
+        ms_type = expr.dtype.element_type[field]
+        is_collection = isinstance(ms_type, hl.tarray) or isinstance(ms_type, hl.tset)
+
+        # Get the highest impact csq label.
+        expr_map = expr.flatmap if is_collection else expr.map
+        filter_func = lambda x, ms: x.contains(ms) if is_collection else x == ms
+        ms_csq = get_most_severe_consequence_expr(
+            expr_map(lambda x: x[field]), csq_order=order
+        )
+
+        # Filter to only consequences with the highest impact csq label, and return
+        # missing if the most severe consequence is missing.
+        return ms_csq, hl.or_missing(
+            (hl.len(expr) > 0) & hl.is_defined(ms_csq),
+            expr.filter(lambda x: filter_func(x[field], ms_csq)),
+        )
+
+    # Initialize the lof struct with missing values.
+    lof_expr = hl.struct(lof=hl.missing(hl.tstr), no_lof_flags=hl.missing(hl.tbool))
```

Comment:
so lof and no_lof_flags will always be None if prioritize_loftee/prioritize_loftee_no_flags are False, even if lof flags are present?

Reply:
I modified this a bit, let me know if the changes make sense